fastapi_events/dispatcher.py

```diff
@@ -116,7 +118,10 @@ def dispatch(
             payload_schema_cls = payload_schema_registry.get(event_name)
             if payload_schema_cls:
                 payload_schema_cls_dict_args = payload_schema_cls_dict_args or DEFAULT_PAYLOAD_SCHEMA_CLS_DICT_ARGS
-                payload = payload_schema_cls(**(payload or {})).dict(**payload_schema_cls_dict_args)
+                if IS_PYDANTIC_V2:
```

Comment:
My sincerest apologies as I completely missed  your contribution. 

Here's my thought on this:
As  `model_dump(...)` is most likely to remain in future versions. It will be more robust if we inverse the if-else logic. I.e.: check if pydantic v1 is used before using `.dict(...)`, and resort to `model_dump(...)` in the else clause.

Once again, my apologies for not seeing your contribution earlier.

Reply:
@melvinkcx changed as requested

Reply:
Thanks, could you please pull the latest changes from the upstream? There has been a new release since your first change.

Reply:
@melvinkcx  rebased and squashed the commits

Reply:
thank you very much!